### PR TITLE
Upgrading flutter, dependencies and frontend workflows

### PR DESCRIPTION
**Pull request by ErwanC7**
Some dependencies can be deleted and a lot of code can be remove but some part REALLY need to be refactor with Dio and BLoC

### CODE REVIEW

File: .github/workflows/backend-linter.yml
Comment (zachmae) [inline .github/workflows/backend-linter.yml:1]: this file shouldn't be part of this pr (no idea how is it consider modified but is ok)

File: .github/workflows/frontend-linter.yml
Comment (zachmae) [inline .github/workflows/frontend-linter.yml:1]: same as backend-linter.yml i guess

**Review by zachmae — COMMENTED** [review]
I don't see any major change that could break the already functionning workflow, it's just modification of the flutter version that i've set on a recent version randomly so LGTM

File: .github/workflows/backend-linter.yml
Comment (ErwanC7) [reply .github/workflows/backend-linter.yml:1]: This file didn't exist. Before, there was a file named "linter.yaml" which contained the front and back linter

File: .github/workflows/frontend-linter.yml
Comment (ErwanC7) [reply .github/workflows/frontend-linter.yml:1]: https://github.com/EIP-GetOut/GETOUT-Platform/pull/25#discussion_r1421160796

**Review by Happinesseuh — APPROVED** [review]
